src/verify.c

```diff
@@ -252,6 +252,7 @@ static int get_required_files(struct manifest *official_manifest, struct list *s
 	if (cmdline_option_install) {
 		progress_set_next_step("download_packs");
 		print("\n");
+		progress_report_interdeterminate();
```

Comment:
so this means you are always going to send a -1 followed by actual percentages when downloading packs and/or fullfiles?

Reply:
Yes, but this does not work really well for downloading packs as that one gets stuck on p_thread_join in last return statement.

So I was initially thinking of not tackling this one.

Reply:
Is it okay for a step to report someting like

{ "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : -1, "stepDescription" : "download_packs" }
{ "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : 12, "stepDescription" : "download_packs" },
{ "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : 25, "stepDescription" : "download_packs" },
{ "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : 37, "stepDescription" : "download_packs" },
{ "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : 50, "stepDescription" : "download_packs" },
{ "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : 62, "stepDescription" : "download_packs" },
 { "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : 75, "stepDescription" : "download_packs" },
.
.
.
.

 { "type" : "progress", "currentStep" : 5, "totalSteps" : 8, "stepCompletion" : 100, "stepDescription" : "download_packs" }

Reply:
my concern with this approach is that if we send a -1 and installer is going to start a spinner based of this value then they won't take advantage of the real percentage values of the download because a spinner is going to be started anyway.... I think we should not send the -1 unless we really know we won't be able to track progress (unless installer can automatically switch from a spinner to a real progress bar as soon as they get a valid positive value)

Reply:
@castulo Good point, but regarding the switching Mark Horn expressed his doubts of effectively switching between the two.

Another approach;
 If we are to do this cleanly, I think we further break a steps like download_packs into sub-steps like pre_precessing packs , download packs, post_processing_packs etc because we are currently doing more than one thing in a given step. download_packs is more a misnomer

Reply:
I have this fixed in my commits